cdc/processor.go

```diff
@@ -1170,24 +1030,57 @@ func (p *processor) sorterConsume(
 				}
 				continue
 			}
-			sinkResolvedTs := atomic.LoadUint64(&p.sinkEmittedResolvedTs)
 			if pEvent.CRTs <= lastResolvedTs || pEvent.CRTs < replicaInfo.StartTs {
 				log.Panic("The CRTs of event is not expected, please report a bug",
 					util.ZapFieldChangefeed(ctx),
 					zap.String("model", "sorter"),
-					zap.Uint64("globalResolvedTs", sinkResolvedTs),
 					zap.Uint64("resolvedTs", lastResolvedTs),
 					zap.Int64("tableID", tableID),
 					zap.Any("replicaInfo", replicaInfo),
 					zap.Any("row", pEvent))
 			}
-			select {
-			case <-ctx.Done():
-				if errors.Cause(ctx.Err()) != context.Canceled {
-					p.errCh <- ctx.Err()
+			failpoint.Inject("ProcessorSyncResolvedError", func() {
+				p.errCh <- errors.New("processor sync resolved injected error")
+				failpoint.Return()
+			})
+			err := processRowChangedEvent(pEvent)
+			if err != nil {
+				if errors.Cause(err) != context.Canceled {
+					p.errCh <- errors.Trace(err)
+				}
+				return
+			}
+		case <-globalResolvedTsReceiver.C:
+			localResolvedTs := atomic.LoadUint64(&p.localResolvedTs)
+			globalResolvedTs := atomic.LoadUint64(&p.globalResolvedTs)
+			var minTs uint64
+			if localResolvedTs < globalResolvedTs {
```

Comment:
In which scenario will `localResolvedTs < globalResolvedTs` happen, do we need to add a warning log here

Reply:
added a warn log